src/main/kotlin/com/openlattice/data/storage/PostgresQueries.kt

```diff
@@ -377,33 +378,43 @@ internal fun selectEntityKeyIdsWithCurrentVersionSubquerySql(
         entitiesClause: String,
         metadataOptions: Set<MetadataOption>,
         linking: Boolean,
+        omitEntitySetId: Boolean,
         joinColumns: List<String>
 ): String {
     val metadataColumns = getMetadataOptions(metadataOptions, linking).joinToString(",")
 
-    var selectColumns = joinColumns.joinToString(",") { column ->
-        if (column == ENTITY_SET_ID.name && linking) {
-            "array_agg(entity_set_id) as entity_set_ids"
-        } else {
-            column
-        }
-    }
-
-    if (metadataColumns.isNotEmpty()) {
-
-        if (linking) {
-            if (metadataOptions.contains(MetadataOption.LAST_WRITE)) {
-                selectColumns += ", max(${LAST_WRITE.name}) AS ${LAST_WRITE.name}"
+    val selectColumns = joinColumns.joinToString(",") +
+            if (metadataColumns.isNotEmpty()) {
+                if (linking) {
+                    if (metadataOptions.contains(MetadataOption.LAST_WRITE)) {
+                        ", max(${LAST_WRITE.name}) AS ${LAST_WRITE.name}"
+                    } else {
+                        ""
+                    } + if (metadataOptions.contains(MetadataOption.ENTITY_SET_IDS)) {
+                        "array_agg(entity_set_id) as entity_set_ids"
+                    } else {
+                        ""
+                    } + if (metadataOptions.contains(MetadataOption.ENTITY_KEY_IDS)) {
```

Comment:
will there ever be multiple of these metadata options passed? if so spacing might be off

Reply:
/commas